report_py3o/py3o_parser.py

```diff
@@ -0,0 +1,217 @@
+# -*- coding: utf-8 -*-
+# Copyright 2013 XCG Consulting (http://odoo.consulting)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+from cStringIO import StringIO
+import json
+import pkg_resources
+import os
+import sys
+from base64 import b64decode
+import requests
+from tempfile import NamedTemporaryFile
+from py3o.template.helpers import Py3oConvertor
+from py3o.template import Template
+from py3o.formats import Formats
+
+from openerp import _
+from openerp import exceptions
+from openerp.report.report_sxw import report_sxw
+from openerp import registry
+
+
+_extender_functions = {}
+
+
+class TemplateNotFound(Exception):
+    pass
+
+
+def py3o_report_extender(report_xml_id=None):
+    """
+    A decorator to define function to extend the context sent to a template.
+    This will be called at the creation of the report.
+    The following arguments will be passed to it:
+        - ir_report: report instance
+        - localcontext: The context that will be passed to the report engine
+    If no report_xml_id is given the extender is registered for all py3o
+    reports
+    Idea copied from CampToCamp report_webkit module.
+
+    :param report_xml_id: xml id of the report
+    :return: a decorated class
+    """
+    global _extender_functions
+
+    def fct1(fct):
+        _extender_functions.setdefault(report_xml_id, []).append(fct)
+        return fct
+    return fct1
+
+
+@py3o_report_extender()
+def defautl_extend(report_xml, localcontext):
+    # add the base64decode function to be able do decode binary fields into
+    # the template
+    localcontext['b64decode'] = b64decode
+
+
+class Py3oParser(report_sxw):
+    """Custom class that use Py3o to render libroffice reports.
+        Code partially taken from CampToCamp's webkit_report."""
+
+    def get_template(self, report_obj):
+        """private helper to fetch the template data either from the database
+        or from the default template file provided by the implementer.
+
+        ATM this method takes a report definition recordset
+        to try and fetch the report template from database. If not found it
+        will fallback to the template file referenced in the report definition.
+
+        @param report_obj: a recordset representing the report defintion
```

Comment:
This is mixing Doxygen and sphinx param format. Prefer sphinx:
```
:param report_obj:
```

Also definition is spelled wrong